### PR TITLE
[REVIEW] Move `cudf._cuda` into `rmm._cuda`

### PR DESCRIPTION
**Pull request by galipremsagar**
Resolves: #488

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/rmm/_cuda/gpu.pyx [hunk 1]
```diff
@@ -275,10 +70,34 @@ def driverGetVersion():
...
+    Set the current CUDA device
+    Parameters
+    ----------
+    device : int
+        The ID of the device to set as current
+    """
+    cdef cudaError_t status = cudaSetDevice(device)
+
+    if status != cudaError.cudaSuccess:
+        raise CUDARuntimeError(status)
```
Comment (kkraus14) [inline python/rmm/_cuda/gpu.pyx:101]: Can we make these more analogous to `cudaSetDevice` / `cudaGetDevice`? i.e. something like `setDevice` / `getDevice` for now?

File: python/rmm/_lib/memory_resource.pyx [hunk 2]
```diff
@@ -329,7 +305,7 @@ cpdef _initialize(
     # Save the current device so we can reset it
     try:
         original_device = get_current_device()
-    except RuntimeError:
+    except CUDARuntimeError as e:
```
Comment (kkraus14) [inline python/rmm/_lib/memory_resource.pyx:315]: Can we catch the actual proper error here now?

Comment (galipremsagar) [reply python/rmm/_lib/memory_resource.pyx:316]: Yes, based on the `status`

**Comment by jakirkham** [comment]
There is also a small amount of generic CUDA stuff in these files. Maybe we want to consolidate these somehow?

https://github.com/rapidsai/rmm/blob/branch-0.15/python/rmm/_lib/lib.pxd
https://github.com/rapidsai/rmm/blob/branch-0.15/python/rmm/_lib/lib.pyx

File: python/rmm/_lib/memory_resource.pyx [hunk 3]
```diff
@@ -313,7 +314,10 @@ cpdef void _initialize(
         if cuda_initialization:
             original_device = get_current_device()
     except CUDARuntimeError as e:
-        raise e
+        if e.status in {cudaError.cudaErrorNoDevice}:
```
Comment (kkraus14) [inline python/rmm/_lib/memory_resource.pyx:317]:
```suggestion
        if e.status == cudaError.cudaErrorNoDevice:
```

**Pull request merged**